### ISSUE TITLE
Add to cart doesn't add Shopify line item properties

### ISSUE THREAD

**Issue by tobymarsden** [ColorlibHQ/Fashe#38]
[This line](https://github.com/ColorlibHQ/Fashe/blob/master/assets/common.js.liquid#L110) doesn't include any line item properties from the product form, which stops the theme from working with all product options and file upload apps, and also makes it difficult to add line item property fields within the theme.

A better approach might be to serialize the form.

**Comment by lexvz** [comment]
Hi there,

I am experiencing the same issue. Did you manage to resolve it?

Thanks!

**Comment by tobymarsden** [comment]
@lexvz Yes, I don't have the code to hand but I just serialized the form (I think `$('#form_buy').serialize()`). This was a store belonging to a customer of my app so I made the minimum changes required to avoid causing any other issues, but it's worked fine for them.

**Comment by puikinsh** [comment]
Fixed in Fashe 2.0, and in the way you suggested.

The add-to-cart request now posts `new FormData(this.form)`, so the whole product form is serialized and every line item property in it is submitted. That makes the theme work with option and file-upload apps, and lets line item property fields be added inside the theme.

Thanks for the report and for diagnosing it precisely — it was a real bug in v1.

2.0.1: https://github.com/ColorlibHQ/Fashe/releases/tag/v2.0.1
